### PR TITLE
CNV-28327: User settings hook infrastracture

### PR DESCRIPTION
**Pull request by metalice**
## 📝 Description

This is the first step of adding the user setting feature.
In this PR I have built the hook that will be in charge of creating/updating the config map of user setting. 
https://issues.redhat.com/browse/CNV-28327
## 🎥 Demo

> Please add a video or an image of the behavior/changes

### CODE REVIEW

File: src/utils/hooks/useUserSettings/useUserSettings.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      setLoading(false);
+    };
+    fetchUserName();
+  }, []);
+
+  const [userConfigMap, loadedConfigMap, configMapError] =
+    useK8sWatchResource<IoK8sApiCoreV1ConfigMap>(
+      userName && {
+        groupVersionKind: modelToGroupVersionKind(ConfigMapModel),
+        namespace: 'default',
```
Comment (avivtur) [inline src/utils/hooks/useUserSettings/useUserSettings.ts:46]: Hi :) can you please use const DEFAULT_NAMESPACE?

File: src/utils/hooks/useUserSettings/useUserSettings.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        name: `${userName}-kubevirt-settings`,
```
Comment (avivtur) [inline src/utils/hooks/useUserSettings/useUserSettings.ts:47]: can you please create a local const for this as it's being used at line 59 as well?

File: src/utils/hooks/useUserSettings/useUserSettings.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  useEffect(() => {
+    if (configMapError?.code === 404 && userName) {
+      setLoading(true);
+      try {
+        k8sCreate({
+          model: ConfigMapModel,
+          data: {
+            metadata: {
+              name: `${userName}-kubevirt-settings`,
+              namespace: 'default',
```
Comment (avivtur) [inline src/utils/hooks/useUserSettings/useUserSettings.ts:60]: here too :)

**Review by avivtur — COMMENTED** [review]
Looks nice :) 
I left a few nits suggestions

File: src/utils/hooks/useUserSettings/useUserSettings.ts (hunk 1 above)
Comment (metalice) [reply src/utils/hooks/useUserSettings/useUserSettings.ts]: done, same for the rest of the comments :)

**Review by avivtur — APPROVED** [review]

**Pull request merged**